daemon/api_test.go

```diff
@@ -360,6 +359,36 @@ var modelDefaults = map[string]interface{}{
 	"kernel":       "kernel",
 }
 
+func (s *apiBaseSuite) fakeServiceControl(st *state.State, appInfos []*snap.AppInfo, inst *servicestate.Instruction, context *hookstate.Context) ([]*state.TaskSet, error) {
+	st.Lock()
+	defer st.Unlock()
```

Comment:
not really related to the PR, but is a bit atypical that servicestate.Control does locking itself instead of the caller, something maybe to consider changing in a follow up, but maybe there's a good reason that I don't remember

Reply:
You're right. Will check in a followup.